### PR TITLE
Fix labels tests

### PR DESCRIPTION
**Pull request by zli82016**
1. fixes https://github.com/hashicorp/terraform-provider-google/issues/16029
   If the labels are in the state with pre-5.0 provider, to delete a label with 5.0 provider, the user has to run `terraform apply` to introduce the `terraform_labels` field with 5.0 provider and then remove the label from the configuration, run `terraform apply`. That is the reason that the test `TestAccDataprocClusterLabelsMigration_withRemoval `failed. I removed this test.
3. Apply new model to alloydb resources
4. Modify the doc for labels/annotations filed

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: mmv1/third_party/terraform/services/dataflow/resource_dataflow_job_test.go.erb [hunk 1]
```diff
@@ -285,7 +285,7 @@ func TestAccDataflowJob_withProviderDefaultLabels(t *testing.T) {
 					resource.TestCheckResourceAttr("google_dataflow_job.with_labels", "terraform_labels.env", "foo"),
 					resource.TestCheckResourceAttr("google_dataflow_job.with_labels", "terraform_labels.default_expiration_ms", "3600000"),
 
-					resource.TestCheckResourceAttr("google_dataflow_job.with_labels", "effective_labels.%", "3"),
+					resource.TestCheckResourceAttr("google_dataflow_job.with_labels", "effective_labels.%", "6"),
```
Comment (rileykarson) [inline mmv1/third_party/terraform/services/dataflow/resource_dataflow_job_test.go.erb:288]: 2 + 3 + 1 system label, right?

File: mmv1/third_party/terraform/services/dataflow/resource_dataflow_job_test.go.erb [hunk 2]
```diff
@@ -365,7 +365,7 @@ func TestAccDataflowJob_withProviderDefaultLabels(t *testing.T) {
 				Config: testAccDataflowJob_zone(bucket, job, zone),
 				Check: resource.ComposeTestCheckFunc(
 					resource.TestCheckNoResourceAttr("google_dataflow_job.with_labels", "labels.%"),
-					resource.TestCheckNoResourceAttr("google_dataflow_job.with_labels", "effective_labels.%"),
+					resource.TestCheckResourceAttr("google_dataflow_job.with_labels", "effective_labels.%", "3"),
```
Comment (rileykarson) [inline mmv1/third_party/terraform/services/dataflow/resource_dataflow_job_test.go.erb:368]: 3 system labels?

File: mmv1/third_party/terraform/services/dataproc/resource_dataproc_cluster_upgrade_test.go [hunk 3]
```diff
@@ -154,59 +154,10 @@ func TestAccDataprocClusterLabelsMigration_withUpdate(t *testing.T) {
...
 					resource.TestCheckNoResourceAttr("google_dataproc_cluster.with_labels", "labels.%"),
-					// We only provide one, but GCP adds three and goog-dataproc-autozone is added internally, so expect 4.
-					resource.TestCheckResourceAttr("google_dataproc_cluster.with_labels", "effective_labels.%", "4"),
-				),
-			},
-		},
-	})
-}
-
-func TestAccDataprocClusterLabelsMigration_withRemoval(t *testing.T) {
```
Comment (rileykarson) [inline mmv1/third_party/terraform/services/dataproc/resource_dataproc_cluster_upgrade_test.go:165]: We can keep the test around for some time, it'll provide a warning if we break an assumption we made in the migration previously.

**Review by rileykarson — APPROVED** [review]

File: mmv1/third_party/terraform/services/dataflow/resource_dataflow_job_test.go.erb (hunk 1 above)
Comment (zli82016) [reply mmv1/third_party/terraform/services/dataflow/resource_dataflow_job_test.go.erb:288]: yes, 2 user defined labels + 1 provider default labels + 3 system labels

File: mmv1/third_party/terraform/services/dataflow/resource_dataflow_job_test.go.erb (hunk 2 above)
Comment (zli82016) [reply mmv1/third_party/terraform/services/dataflow/resource_dataflow_job_test.go.erb:368]: right.

File: mmv1/third_party/terraform/services/dataproc/resource_dataproc_cluster_upgrade_test.go (hunk 3 above)
Comment (zli82016) [reply mmv1/third_party/terraform/services/dataproc/resource_dataproc_cluster_upgrade_test.go:165]: Sounds good.

**Pull request merged**